tests/testDipoleFitter.py

```diff
@@ -0,0 +1,407 @@
+        dm = di.getMask()
+        posDetectedBits = posImage.getMaskedImage().getMask().getArray() == dm.getPlaneBitMask("DETECTED")
+        negDetectedBits = negImage.getMaskedImage().getMask().getArray() == dm.getPlaneBitMask("DETECTED")
+        pos_det = dm.addMaskPlane("DETECTED_POS")  # new mask plane -- different from "DETECTED"
+        neg_det = dm.addMaskPlane("DETECTED_NEG")  # new mask plane -- different from "DETECTED_NEGATIVE"
+        dma = dm.getArray()
+        # set the two custom mask planes to these new masks
+        dma[:, :] = posDetectedBits * pos_det + negDetectedBits * neg_det
+        return dipole, (posImage, posCatalog), (negImage, negCatalog)
+
+    @staticmethod
+    def detectDipoleSources(diffim, doMerge=True, detectSigma=5.5, grow=3):
+        """
+        Utility function for detecting dipoles. Detects pos/neg sources in the diffim,
+        then merges them. A bigger "grow" parameter leads to a larger footprint which
+        helps with dipole measurement for faint dipoles.
+        """
+
+        # Start with a minimal schema - only the fields all SourceCatalogs need
+        schema = SourceTable.makeMinimalSchema()
+
+        # Customize the detection task a bit (optional)
+        detectConfig = SourceDetectionConfig()
+        detectConfig.returnOriginalFootprints = False  # should be the default
+
+        psfSigma = diffim.getPsf().computeShape().getDeterminantRadius()
+
+        # code from imageDifference.py:
+        detectConfig.thresholdPolarity = "both"
+        detectConfig.thresholdValue = detectSigma
+        # detectConfig.nSigmaToGrow = psfSigma
+        detectConfig.reEstimateBackground = True  # if False, will fail often for faint sources on gradients?
+        detectConfig.thresholdType = "pixel_stdev"
+
+        # Create the detection task. We pass the schema so the task can declare a few flag fields
+        detectTask = SourceDetectionTask(schema, config=detectConfig)
+
+        table = SourceTable.make(schema)
+        catalog = detectTask.makeSourceCatalog(table, diffim, sigma=psfSigma)
+
+        # Now do the merge.
+        if doMerge:
+            fpSet = catalog.fpSets.positive
+            fpSet.merge(catalog.fpSets.negative, grow, grow, False)
+            sources = SourceCatalog(table)
+            fpSet.makeSources(sources)
+
+            return sources
+
+        else:
+            return detectTask, schema
+
+
+def suite():
+    """Returns a suite containing all the test cases in this module."""
+
+    lsst_tests.init()
+
+    suites = []
+    suites += unittest.makeSuite(DipoleFitAlgorithmTest)
+    suites += unittest.makeSuite(DipoleFitTaskTest)
+    suites += unittest.makeSuite(DipoleFitTaskEdgeTest)
+    suites += unittest.makeSuite(lsst_tests.MemoryTestCase)
+    return unittest.TestSuite(suites)
+
+
+def run(shouldExit = False):
+    """Run the tests"""
+    lsst_tests.run(suite(), shouldExit)
```

Comment:
OK, done, although it only reports 6 tests completed rather than 7 when just running the script.
```
% py.test tests/testDipoleFitter.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.11, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /Users/dreiss/GIT_REPOS/ip_diffim/tests, inifile:
collected 6 items

tests/testDipoleFitter.py ......

==================================================== 6 passed in 2.06 seconds ====================================================
```